pkg/workloads/cortex/client/cortex/client.py

```diff
@@ -173,42 +182,45 @@ def _deploy(
 
         deploy_results = json.loads(output.strip())
 
+        deploy_result = deploy_results[0]
+
         if not wait:
-            return deploy_results
+            return deploy_result
 
         def stream_to_stdout(process):
             for c in iter(lambda: process.stdout.read(1), ""):
                 sys.stdout.write(c)
-
-        for deploy_result in deploy_results:
-            api_name = deploy_result["api"]["spec"]["name"]
-            kind = deploy_result["api"]["spec"]["kind"]
-            if kind != "RealtimeAPI":
-                continue
-
-            env = os.environ.copy()
-            env["CORTEX_CLI_INVOKER"] = "python"
-            process = subprocess.Popen(
-                [get_cli_path(), "logs", "--env", self.env, api_name],
-                stderr=subprocess.STDOUT,
-                stdout=subprocess.PIPE,
-                encoding="utf8",
-                env=env,
-            )
-
-            streamer = threading.Thread(target=stream_to_stdout, args=[process])
-            streamer.start()
-
-            while process.poll() is None:
-                api = self.get_api(api_name)
-                if api["status"]["status_code"] != "status_updating":
-                    if api["status"]["status_code"] == "status_live":
-                        time.sleep(2)
-                    process.terminate()
-                    break
-                time.sleep(2)
-
-        return deploy_results
+                sys.stdout.flush()
+
+        api_name = deploy_result["api"]["spec"]["name"]
+        kind = deploy_result["api"]["spec"]["kind"]
+        if kind != "RealtimeAPI":
```

Comment:
Thinking whether getting rid of `kind` var is good or bad. What do you think? It would look like `if deploy_result["api"]["spec"]["kind"] != "RealtimeAPI"`.